### PR TITLE
[mongodb] MongoError constructor accept string and object

### PR DESCRIPTION
**Pull request by mathieug**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [ ] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: http://mongodb.github.io/node-mongodb-native/3.5/api/MongoError.html
- [ ] If this PR brings the type definitions up to date with a new version of the JS library, update the version number in the header.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`. If for reason the any rule need to be disabled, disable it for that line using `// tslint:disable-next-line [ruleName]` and not for whole package so that the need for disabling can be reviewed.

### CODE REVIEW

File: types/mongodb/index.d.ts [hunk 1]
```diff
@@ -230,11 +230,11 @@ export type WithTransactionCallback<T> = (session: ClientSession) => Promise<T>;
  * see {@link http://mongodb.github.io/node-mongodb-native/3.5/api/MongoError.html}
  */
 export class MongoError extends Error {
-    constructor(message: string);
+    constructor(message: string | object);
```
Comment (CaselIT) [inline types/mongodb/index.d.ts]: It seems to accept also an `Error`, maybe we could add it to the types?

**Review by CaselIT — CHANGES_REQUESTED** [review]

File: types/mongodb/index.d.ts (hunk 1 above)
Comment (mathieug) [reply types/mongodb/index.d.ts]: To me, an instance of `Error` in an `object`.
[TypeScript documentation](https://www.typescriptlang.org/docs/handbook/basic-types.html#object):
> object is a type that represents the non-primitive type, i.e. anything that is not number, string, boolean, bigint, symbol, null, or undefined.

Comment (CaselIT) [reply types/mongodb/index.d.ts]: Yes, but it's more readable by an user if also Error is listed. I don't think typescript will omit it from the intellisense even if its technically superseded by object

Comment (mathieug) [reply types/mongodb/index.d.ts]: Done :)

**Review by CaselIT — APPROVED** [review]

**Comment by mathieug** [comment]
I'll let @RyanCavanaugh investigate and update this PR, thanks.

**Comment by peterblazejewicz** [comment]
> I'll let ... investigate and update this PR, thanks.

There is something wrong with TS@next and existig defintions overriding/extending Promise like interface. See details of the CI, I've noted this in other failing CI. Hope I'll have an update on this shortly.

**Comment by peterblazejewicz** [comment]
@mathieug it's just surfaced:
https://github.com/microsoft/TypeScript/issues/37556

**Comment by peterblazejewicz** [comment]
@mathieug 
You should be able to run local `npm test` now to verify status and push update (or empty commit) to restart CI:
https://github.com/microsoft/TypeScript/pull/37610

**Comment by DanielRosenwasser** [comment]
Really this is just the same as `string | object`

**Pull request merged**